### DISCUSSION TITLE
Pest testing Flux::toast

### DISCUSSION THREAD

**Discussion by codeldev** [livewire/flux#427] [Feature Requests] [answered]
Hi all,

Trying to update my Livewire based pest tests for Flux::toast, but not having much luck.

Previously I had my own notifications implemented and could test these by using:

```
 ->assertDispatched('notify',
     type       : 'success',
     redirect : null,
     content  : trans('config.branding.logo.uploaded')
)
```

I can test that a toast was triggered via:

```
->assertDispatched('toast-show')
```

But I would like to ensure the right message was triggered

```
 ->assertDispatched('toast-show',
     variant  : 'success',
     text : trans('config.branding.logo.uploaded')
);
```

Unfortunately the above always fails. I've looked at the toast js file in the vendor directory, but not sure where I'm going wrong. 

Any help appreciated. Many thanks

**Comment by codeldev** [comment, marked as answer]
For anyone else struggling with this, after trial & error, this worked for me:

```
->assertDispatched('toast-show',
    duration: 3000,
    slots   : [
        'text' => trans('config.branding.logo.uploaded'),
    ],
    dataset : [
        'variant'   => 'success',
        'position' => 'top right',
    ]
)
```

**Comment by troccoli** [comment]
Just a heads-up for other people coming here.

If you have specified both some text and a heading, like for example

```php
Flux::toast(
    heading: 'Error!',
    text: 'The credentials provided do not match our records.',
    variant: 'danger',
);
```

Then the order on the items in the `slots` array is important! It's always `text` first and then `heading`

```
assertDispatched(
    event: 'toast-show',
    slots: [
        'text' => 'The credentials provided do not match our records.',
        'heading' => 'Error!',
    ],
    dataset: [
        'variant' => 'danger',
    ],
);
```

Similarly for the `dataset`: it's always `variant` first and then `position`.

See `vendor/livewire/flux/src/Concerns/InteractsWithComponents.php:78`. This for Flux 1.0.18

**Reply by raz-iacob** [reply]
How would you do it for modals?

**Reply by raz-iacob** [reply]
Ok.. I figure it out. for others that want to test modals here is what I did

```php
assertDispatched(
      event: 'modal-close', 
      name: 'add-new'
)
```
